src/R4Mvc.Tools/CodeGen/MethodBuilder.cs

```diff
@@ -15,7 +15,7 @@ public class MethodBuilder
         private IList<ParameterSyntax> _parameters = new List<ParameterSyntax>();
         private BlockSyntax _bodyBlock;
         private ExpressionSyntax _expressionBodySyntax;
-        private bool _useGeneratedAttributes = false, _useNonActionAttribute = false, _noBody = false;
+        private bool _useGeneratedAttributes = false, _useNonActionAndNonHandlerAttributes = false, _useNonHandlerAttribute = false, _noBody = false;
```

Comment:
Why force adding the `[NonHandler]` attribute everywhere you add `[NonAction]`?

As far as I understand, `[NonAction]` should be used in the MVC generated classes, while `[NonHandler]` should be used on RazorPages. Could you confirm if I'm right, and update the PR accordingly then?

Reply:
Yep, thought so - will do